### DISCUSSION TITLE
Whether I should use Vercel and API routes to accomplish my goals.

### DISCUSSION THREAD

**Discussion by Djboy08** [vercel/community#386] [Help] [answered]
Hi! I had a quick question I hope someone could help me with.

My question is in regards to next.js and maybe Vercel. If I for example want to host the app on Vercel but I do know my project is going to have maybe 10k+ requests a second (arbitrary amount), should I look into other platforms, such as using a Custom server for next.js?

**Comment by ChristianDavis** [comment]
10k+ requests a second is an extremely high volume amount to pull out of the air. Are you looking for general decision criteria? Have you read the conceptual model for serverless functions? https://vercel.com/docs/concepts/functions/conceptual-model

If the functions are stateless, they can generally scale horizontally constrained by how much you're willing to pay for compute. If they need to contain state (like connections to a DB), then you incur tradeoffs with connection time and hit bottlenecks with concurrent connections to the DB.

**Reply by samcx** [reply]
Vercel automatically scales traffic automatically for static files, but there is indeed a Function concurrency limit → https://vercel.com/docs/concepts/limits/overview#serverless-function-concurrency. Proper caching of Functions go a long way to not reach this limit, and using static where you can.

Is this 10k traffic a mix of both static and Serverless?

**Comment by wiiiimm** [comment, marked as answer]
@Djboy08 Are your 10k+ requests all uncached and are all requesting for serverless functions at `/api/...`? As @samsisle pointed out in https://github.com/vercel/community/discussions/386#discussioncomment-2370884, we have a 1000 concurrent request limit on our serverless functions for hobby and Pro plans. If you need to support 10K+ API requests per second, you can sign up to Vercel's Enterprise plan over at https://vercel.com/pricing.
